GeneratorInterface/LHEInterface/plugins/ExternalLHEProducer.cc

```diff
@@ -225,7 +225,8 @@ ExternalLHEProducer::beginRunProduce(edm::Run& run, edm::EventSetup const& es)
   
   std::ostringstream eventStream;
   eventStream << nEvents_;
-  args_.push_back(eventStream.str());
```

Comment:
The string vector args, defined in the job cfg, conventionally only contains the gridpack path (P), so args = [P].
Currently ExternalLHEProducer appends the number of events (N), random seed (S) and number of threads (T) arg,
producing therfore args* = [P,N,S,T] which is expected by run_generic_tarball_cvmfs.sh in this order.

If one adds other parameters (O) to args in the cfg,  args = [P,O] and args* = [P,O,N,S,T] therefore run_generic_tarball_cvmfs.sh breaks.
What I did was to enforce args* = [P,N,S,T,O] in ExternalLHEProducer keeping back-compatibility in run_generic_tarball_cvmfs.sh, further allowing O to be passed (whenever present) to runcmsgrid.sh with
./runcmsgrid.sh $nevt $rnum $ncpu ${@:5}
where ${@:5} means "all the arguments passed to the script starting from the 5th, i.e. O"

I tested and seems to work flawless, I hope is clear.
Let me know if you have further comments or questions